### ISSUE TITLE
How much disk does one need to run this ?

### ISSUE THREAD

**Issue by sokoow** [leblancd/conKube#1]
This system is absolutely huge - I'm giving it 30GB of disk space and it's not enough :D How much disk space this thing needs ? Container that is being created is 10+gigs itself, it's madness :D

**Comment by leblancd** [comment]
@sokoow - Thanks for trying this out!
Are you doing a docker run with "--tmpfs /docker-graph:exec"? I added this flag a couple of days ago. The container should run much faster, and the disk usage has gone down significantly. You will need 12 GB of system memory (maybe less with swaps, I'm experimenting with that now).

**Comment by sokoow** [comment]
Okay, that works now, thanks :) The only challenge I have right now is that I'm trying to redirect internal ipv6 addresses to my real host addresses, and I'm stuck a bit. Also, it'd be nice to have some pre-caching of these docker images that are being pulled all the time - they're quite big.

**Comment by leblancd** [comment]
@sokoow - Great to hear the disk space isn't a stopper now. Can you tell me more about redirecting the internal IPv6 addresses... do you mean you're trying to route packets from outside the cluster that are destined for the pod network?

**Comment by sokoow** [comment]
Yeah, I'm basically trying to pull out a nodeport from one of the kube 'nodes' out to the external world/IP both on ipv4 and ipv6 - you know how to do that ?

I'm also starting to wonder on what is taking so much space in this whole setup ? https://hub.docker.com/r/diverdane/kube-in-the-box/ takes about 1.9GB, and it pulls a lot more. I can't introspect this image a lot, cause ... I can't find a dockerfile anywhere for it :/

**Comment by leblancd** [comment]
@sokoow - For dual-stack access from the external world, nodeport won't work at the moment... that will have to wait until Kubernetes nodeports are dual-stack. (The current proposal is here, but it's in flux at the moment: https://github.com/kubernetes/community/blob/fc4d40bac4ca76d9c46fa7335ea74a186388c313/keps/sig-network/0013-20180612-ipv4-ipv6-dual-stack.md#proposal).

What *is* possible now is to expose your services via an NGINX ingress controller running on the node's host network. That will require setting up dual-stack addresses on the nodes. You can find instructions on how to do that here: https://github.com/leblancd/kube-v6/tree/master/dual-stack-ingress#installing-a-dual-stack-ingress-controller-on-an-ipv6-only-kubernetes-cluster

I've added documentation on the dockerhub page to reduce the mystery :) (copied from this github repo):  https://hub.docker.com/r/diverdane/kube-in-the-box/

One easy reduction in size would be to drop the kubekins-e2e baseline image, and use the bootstrap image instead (see https://github.com/leblancd/kube-in-the-box#kube-in-the-box) . I don't know how much more can be reduced beyond that, the K-D-C scripts build a massive Kubernetes "hyperkube" binary (a multi-purpose executable used for the API server, controller manager, kubelet, etc.), and then it builds a "hypokube" docker image that is the image used for container nodes.

I really appreciate your trying this out, and your helpful suggestions.
